appengine/genetics/js/blocks.js

```diff
@@ -0,0 +1,465 @@
+    var thisBlock = this;
+    this.setTooltip(function() {
+      var mode = thisBlock.getFieldValue('PROPERTY');
+      var TOOLTIPS = {
+        'SIZE': BlocklyGames.getMsg('Genetics_sizeTooltip'),
+        'AGGRESSIVENESS': BlocklyGames.getMsg('Genetics_aggressivenessTooltip'),
+        'FERTILITY': BlocklyGames.getMsg('Genetics_fertilityTooltip'),
+        'START_FERTILITY': BlocklyGames.getMsg('Genetics_startFertilityTooltip'),
+        'SEX': BlocklyGames.getMsg('Genetics_sexTooltip'),
+        'AGE': BlocklyGames.getMsg('Genetics_ageTooltip'),
+        'ID': BlocklyGames.getMsg('Genetics_idTooltip'),
+        'PICK_FIGHT': BlocklyGames.getMsg('Genetics_pickFightOwnerTooltip'),
+        'CHOOSE_MATE': BlocklyGames.getMsg('Genetics_chooseMateOwnerTooltip'),
+        'MATE_ANSWER': BlocklyGames.getMsg('Genetics_mateAnswerOwnerTooltip')
+      };
+      return TOOLTIPS[mode];
+    });
+  },
+  /**
+   * Modify this block to have the correct output type.
+   * @param {string} newProp Either a property requesting a function owner that
+   * returns a string or a property that returns a number.
+   * @private
+   * @this Blockly.Block
+   */
+  updateType_: function(newProp) {
+    if (newProp == 'SEX' || newProp == 'PICK_FIGHT' ||
+        newProp == 'CHOOSE_MATE' || newProp == 'MATE_ANSWER') {
+      this.outputConnection.setCheck('String');
+    } else {
+      this.outputConnection.setCheck('Number');
+    }
+  }
+};
+
+/**
+ * Defines the JavaScript generation for getProperties.
+ * @param {Blockly.Block} block
+ * @return {!Array.<string|number>}
+ */
+Blockly.JavaScript['genetics_getProperties'] = function(block) {
+  // Generate JavaScript for getting mouse property.
+  var mouse = Blockly.JavaScript.valueToCode(block, 'MOUSE',
+          Blockly.JavaScript.ORDER_NONE) || 'me()';
+  var property = block.getFieldValue('PROPERTY');
+  var code = mouse + '.';
+  switch (property) {
+    case 'SIZE':
+      code += 'size';
+      break;
+    case 'AGGRESSIVENESS':
+      code += 'aggressiveness';
+      break;
+    case 'START_AGGRESSIVENESS':
+      code += 'aggressiveness';
+      break;
+    case 'FERTILITY':
+      code += 'fertility';
+      break;
+    case 'START_FERTILITY':
+      code += 'startFertility';
+      break;
+    case 'SEX':
+      code += 'sex';
+      break;
+    case 'AGE':
+      code += 'age';
+      break;
+    case 'ID':
+      code += 'id';
+      break;
+    case 'PICK_FIGHT':
+      code += 'pickFightOwner';
+      break;
+    case 'CHOOSE_MATE':
+      code += 'chooseMateOwner';
+      break;
+    case 'MATE_ANSWER':
+      code += 'mateAnswerOwner';
+      break;
+    default:
+      throw 'Unknown math operator: ' + operator;
+  }
+  return [code, Blockly.JavaScript.ORDER_MEMBER];
+};
+
+/**
+ * Block for getting the enum for mouse sexes.
+ * @this Blockly.Block
+ */
+Blockly.Blocks['genetics_sex'] = {
+  /**
+   * Block for getting mouse sexes
```

Comment:
All possible sexes, or the sex for a particular mouse? Documentation is a bit unclear.

"Block for defining the sex of a mouse", perhaps (IIUC)?

Also, no need to replicate this same docstring above the function (lines 373-376).